app/policies/article_policy.rb

```diff
@@ -17,9 +17,9 @@ def create?
   end
 
   def edit?
-    if @record.publication_id
+    if @record.publication_id && @user
       @record.publication.user_publications.where(user_id: @user.id).any? && @record.user_id == @user.id
-    else
+    elsif @user
```

Comment:
what happens when there is no `@user` variable? E.g. when the user is not logged in.